src/com/palmergames/bukkit/towny/TownyFormatter.java

```diff
@@ -783,13 +783,15 @@ private static void addTownMoneyComponents(Town town, Translator translator, Sta
 				town.getAccount().setDebtCap(MoneyUtil.getEstimatedValueOfTown(town));
 			screen.addComponentOf("bankrupt", translator.of("status_bank_bankrupt") + " " + colourKeyValue(translator.of("status_debtcap"), "-" + formatMoney(town.getAccount().getDebtCap())));
 		}
-		if (town.hasUpkeep())
+		if (town.hasUpkeep() && TownySettings.isTaxingDaily())
 			screen.addComponentOf("upkeep", translator.of("status_splitter") + colourKey(translator.of("status_bank_town2")) + " " + colourKeyImportant(formatMoney(BigDecimal.valueOf(TownySettings.getTownUpkeepCost(town)).setScale(2, RoundingMode.HALF_UP).doubleValue())));
 		if (TownySettings.getUpkeepPenalty() > 0 && town.isOverClaimed())
 			screen.addComponentOf("upkeepPenalty", translator.of("status_splitter") + colourKey(translator.of("status_bank_town_penalty_upkeep")) + " " + colourKeyImportant(formatMoney(TownySettings.getTownPenaltyUpkeepCost(town))));
 		if (town.isNeutral() && TownySettings.getTownNeutralityCost() > 0)
 			screen.addComponentOf("neutralityCost", translator.of("status_splitter") + colourKey(translator.of("status_neutrality_cost") + " " + colourKeyImportant(formatMoney(TownySettings.getTownNeutralityCost()))));
-		screen.addComponentOf("towntax", translator.of("status_splitter") + colourKey(translator.of("status_bank_town3")) + " " + colourKeyImportant(town.isTaxPercentage() ? town.getTaxes() + "%" : formatMoney(town.getTaxes())));
+		
+		if (TownySettings.isTaxingDaily())
+			screen.addComponentOf("towntax", translator.of("status_splitter") + colourKey(translator.of("status_bank_town3")) + " " + colourKeyImportant(town.isTaxPercentage() ? town.getTaxes() + "%" : formatMoney(town.getTaxes())));

```

Comment:
The isTaxingDaily() setting actually controls all four of these costs being charged and should probably be wrapping all of them.